internal/datadog/metrics_client.go

```diff
@@ -0,0 +1,97 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadog // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog"
+
+import (
+	"context"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type metricsClient struct {
+	meter  metric.Meter
+	gauges map[string]float64
+	mutex  sync.Mutex
+}
+
+// NewMetricClient returns a new metrics client
+func NewMetricClient(meter metric.Meter) metrics.StatsClient {
+	// We need to lock the mutex here to avoid
+	// duplicate metrics being created by both datadogconnector and exporter
+	var mutex sync.Mutex
+	mutex.Lock()
+	defer mutex.Unlock()
+	m := &metricsClient{
+		meter:  meter,
+		gauges: make(map[string]float64),
+	}
+	metrics.Client = m
+	return m
+}
+
+func (m *metricsClient) Gauge(name string, value float64, tags []string, _ float64) error {
+	m.mutex.Lock()
```

Comment:
You either need to use `defer` here
```suggestion
	m.mutex.Lock()
	defer m.mutex.Unlock()
```

or unblock before the `return nil` below